### PR TITLE
checkpointing fix for mixtral

### PR DESCRIPTION
**Pull request by skaulintel**
same as https://github.com/HabanaAI/optimum-habana-fork/pull/215 but for the habana-main branch

### CODE REVIEW

File: optimum/habana/checkpoint_utils.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def model_on_meta(config):
     """
     Checks if load the model to meta.
     """
-    return config.model_type in ["bloom", "llama", "falcon"]
+    return config.model_type in ["bloom", "llama", "falcon", "mixtral"]
```
Comment (jiminha) [inline optimum/habana/checkpoint_utils.py:96]: @skaulintel  what is this change? Do we need to add Mistral as well?

Comment (skaulintel) [reply optimum/habana/checkpoint_utils.py:96]: probably, @libinta can you confirm?

**Review by libinta — APPROVED** [review]

**Comment by skaulintel** [comment]
OH upstream: [https://github.com/huggingface/optimum-habana/pull/909/files]( https://github.com/huggingface/optimum-habana/pull/909)

File: optimum/habana/checkpoint_utils.py (hunk 1 above)
Comment (skaulintel) [reply optimum/habana/checkpoint_utils.py:96]: actually not need for mistral @jiminha

**Pull request merged**